helper/schema/schema.go

```diff
@@ -1365,11 +1365,26 @@ func (m schemaMap) validate(
 			"%q: this field cannot be set", k)}
 	}
 
-	if raw == config.UnknownVariableValue {
-		// If the value is unknown then we can't validate it yet.
-		// In particular, this avoids spurious type errors where downstream
-		// validation code sees UnknownVariableValue as being just a string.
-		return nil, nil
+	// If the value is unknown then we can't validate it yet.
+	// In particular, this avoids spurious type errors where downstream
+	// validation code sees UnknownVariableValue as being just a string.
+	switch raw := raw.(type) {
+	case string:
+		if raw == config.UnknownVariableValue {
+			return nil, nil
+		}
+	case []interface{}:
```

Comment:
Would this also trip up on a list of lists where the second level contains unknowns? I don't know that any provider has ever actually done that, but I think the schema would allow it. :thinking: 

I kinda remember seeing a helper function somewhere to recursively visit all of a data structure looking for `config.UnknownVariableValue`, which in our new world in Terraform Core was of course replaced by `cty`'s `IsWhollyKnown`. I wonder if we still have that old-school helper around somewhere to use here. HIL is the most likely place I'd expect to find it, since it would've needed it for its special behavior of flattening a collection containing unknowns down into a single unknown.

Reply:
Yeah, I forgot that helper/schema accepts nested types in places, so just made the filter recursive.

There is a similar check already (probably multiple, and `UnknownAsNull` in cty), but the schema needs to have a value for required fields in order to validate, so we have to filter them later.